app/controllers/checkouts_controller.rb

```diff
@@ -0,0 +1,78 @@
+class CheckoutsController < ApplicationController
+  before_action :set_variables, only: [:checkout_current_card, :checkout_new_card]
+
+
+  def new
+    @product = Product.find(params[:id])
+    @cards = Card.where(user: current_user)
+  end
+
+
+  def checkout_current_card
+    @charge = Stripe::Charge.create(
```

Comment:
Could we move the common functions on this and the `checkout_new_card` method to a common function that they both call?